ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableMetaRefSpec.java

```diff
@@ -20,21 +20,30 @@
 
 import com.google.common.base.MoreObjects;
 
-public class AlterTableBranchSpec<T> {
+public class AlterTableMetaRefSpec<T> {
```

Comment:
should we mark it as abstract and extend CreateTagSpec and CreateBranchSpec from it? operationType could be set in the constructor

Reply:
I have replaced `CreateBranchSpec `& `CreateTagSpec ` with one inner class `CreateMetaRefSpec`. I think this will be more simpler. Fixed in https://github.com/apache/hive/pull/4372/commits/a63d3708d75a577d1f58a72455e2516a80a2fa09
Please take a look again. Thanks.

Reply:
why do you need `AlterTableMetaRefSpec` wrapper at all, can't we just use parametrized CreateMetaRefSpec<T>
new CreateMetaRefSpec<MetaRef.BRANCH>?

Reply:
I want to use `AlterTableMetaRefSpec `to wrap multiple branch/tag operation, e.g. `CreateMetaRefSpec `as well as the following task `ReplaceMetaRefSpec`, and maybe other operations.
I will remove the wrapper if these operation have same fields in the future work and will change this as you sugested.